### PR TITLE
feat: center the grids as in the old deeprank

### PR DESCRIPTION
**Pull request by cbaakman**
Grids must be given a center position on creation.

With this modification, the query object assigns a center position to the graph objects. For single residue variant graphs, this is the variant residue position. For protein-protein interaction graphs, this is the mean of the contacting atom positions.

When the graph is converted into a grid, this center position is passed on and it will determine the position of the grid points.

A test has been added to check that the center position is the same as in the original deeprank.

### CODE REVIEW

**Review by DaniBodor — CHANGES_REQUESTED** [review]
Can you fix the checks before review :)

**Comment by cbaakman** [comment]
Checks have passed

File: deeprankcore/utils/graph.py [hunk 1]
```diff
@@ -99,6 +99,9 @@ def __init__(self, id_: str):
         # targets are optional and may be set later
         self.targets = {}
 
+        # the center only needs to be set when this graph should be mapped to a grid.
+        self.center = np.array((0.0, 0.0, 0.0))
+
```
Comment (DaniBodor) [inline deeprankcore/utils/graph.py:104]: Then why does the Graph class need a center attribute?

**Review by DaniBodor — APPROVED** [review]
Looks good to me. Just one comment about something that's unclear to me, but doesn't really matter.

**Review by gcroci2 — APPROVED** [review]
Seems good to me! Only a comment about inputs for GridSettings, I remember we agreed on using lists that then should be converted to numpy arrays internally.

File: deeprankcore/utils/graph.py (hunk 1 above)
Comment (cbaakman) [reply deeprankcore/utils/graph.py:104]: That's right! They will be overwritten once the center is known. The problem is that this center position can only be determined from the atoms in the structure. The method `write_as_grid_to_hdf5` only takes a graph as input, not a structure. So I chose to store the center in the graph, just like the node positions.

I agree that this is not a clean solution. Should we redesign the class-diagram?

**Comment by cbaakman** [comment]
> Seems good to me! Only a comment about inputs for GridSettings, I remember we agreed on using lists that then should be converted to numpy arrays internally.

I can change this, no problem.

> Also, to be sure, the centering will work for both ppi residue level and ppi atomic level, right?

Yes, both query object types have the centering implemented. They both center on the mean position of the contact atoms.

File: tests/utils/test_graph.py [hunk 2]
```diff
@@ -61,7 +61,7 @@ def test_graph_build_and_export(): # pylint: disable=too-many-locals
         graph.write_to_hdf5(hdf5_path)
 
         # export grid to hdf5
-        grid_settings = GridSettings(np.array((0, 0, 0)), np.array((20, 21, 21)), np.array((20.0, 21.0, 21.0)))
+        grid_settings = GridSettings(np.array((20, 21, 21)), np.array((20.0, 21.0, 21.0)))
```
[the remark this answers is not in the document]
Comment (cbaakman) [reply tests/utils/test_graph.py:64]: I will make it lists.

**Review by DaniBodor — APPROVED** [review]

**Pull request merged**